bite-size-notes/week-4-b-discrete-LQR.md

```diff
@@ -0,0 +1,197 @@
+
+This fact leads to Bryson's Rule [4], which is a popular method for determining a starting point for $Q$ and $R$. In fact, MATLAB's example documentation on LQR defaults on using Bryson's rule to choosing $Q$ and $R$ [5]. Bryson's Rule states that, for diagonal matricies $Q$ and $R$, the diagonal elements are chosen such that:
+
+$$
+\begin{split}
+Q_{ii} &= \frac{1}{\text{maximum allowable value of }x_i} \\
+R_{ii} &= \frac{1}{\text{maximum allowable value of }u_i}
+\end{split}
+$$
+
+Using Bryson's rule scales the maximum value of each term that appears in the cost function to 1. This can be especially important when state/control inputs are in different units, or are of vastly different magnitudes. This allows the weighting for each state and control term to be roughly comparable, which can be helpful as a starting point as the contributions of each state and control input to the cost function can be more easily compared.
+
+Of course, Bryson's rule is just an elementary method of choosing weights. There is no definite systematic method for choosing weights, but there are interesting methods to explore, such as the algebraic approach outlined in [6] in which the time domain design specifications of the system are directly synthesized into the cost function, and therefore the weights are chosen in direct relation to the control objective.
+
+
+## Conclusion
+
+In the above, we have shown how the discrete time LQR problem is derived and solved for both the finite and infinite horizon case. In doing so, we notice the following advantages and disadvantages of using LQR to solve an optimal control problem:
+
+#### Pros:
+1. **Abstracted Design**: The LQR problem allows for the design of the control input to be abstracted from the physical intuition or understanding of the system being controlled. This can be especially useful for complex systems where physical intuition may be difficult to obtain.
+2. **Efficiency**: The finite horizonLQR problem can be solved in $O(Kn^3)$ time using dynamic programming, which can significantly faster than brute force or naive methods. The infinite horizon case can be solved in constant time using the DARE. Both methods are also efficient to implement in code.
+3. **Optimal Control**: The LQR problem is convex and guarantees that the control input $u^*$ is globally optimal for the cost function at each time step $k$.
```

Comment:
hmm, LQR problem is convex but does the certain constraints or dynamics introduce non-convexities?
or are we assuming that the constraints are also possessing convexity

Reply:
The LQR problem does not contain any constraints, and dynamics are either assumed to be or already linearized as we can describe our system as $x_{k+1} = Ax_k + Bu_k$. The state and control constraints would be part of this linearization and represented in matrices $A$ and $B$.

Your comment does bring up a good point, I use the term "globally optimal"  a little loosely here. Will clarify that it's globally optimal for the LQR formulation, which is not necessarily the full state global optimal, but the point of LQR is that we can avoid doing full state feedback. Thanks for bringing this up!